### PR TITLE
Improvements to app documentation

### PR DESCRIPTION
**Pull request by mikewilburn**
This PR:
- renames `docs/index.md` to `docs/README.md`
- adds table of contents to (2) README files. The TOC was generated using MDTOC.
- links to `docs/README.md` from root README.md
- describes patch changes in RELEASE.md
- increments app version patch number

@nCastle1 I'm less familiar with versioning in this project, so let me know if I missed any files in 503c4a3 where I should have also incremented the app version.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,5 +1,75 @@
 # Data Collection for .NET documentation
```
Comment (esreli) [inline docs/README.md:1]: We might want to keep this consistent with other OSA. What do you think?

```suggestion
# Contents
```

**Review by esreli — APPROVED** [review]
Leaving only a comment was not my intention in the last review, my mistake. Approved!

**Comment by mikewilburn** [comment]
Thanks @esreli. I just pushed up one more commit to fix a hyperlink so that it now actively redirects. Does that approach seem sensible to you?

**Review by esreli — APPROVED** [review]
Nice choice 👍

**Review by nCastle1 — APPROVED** [review]

**Pull request merged**